core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -109,6 +113,121 @@ public DefaultRequestLog(RequestContext ctx) {
         this.ctx = requireNonNull(ctx, "ctx");
     }
 
+    @Override
+    public void addChild(RequestLog child) {
+        checkState(!hasLastChild, "last child is already added");
+        requireNonNull(child, "child");
+        if (children == null) {
+            // first child's all request-side logging events are propagated immediately to the parent
+            children = new ArrayList<>();
+            propagateRequestSideLog(child);
+        }
+        children.add(child);
+    }
+
+    private void propagateRequestSideLog(RequestLog child) {
+        child.addListener(log -> {
+            final long requestStartTimeNanos;
+            final long requestStartTimeMillis;
+            if (log instanceof DefaultRequestLog) {
+                requestStartTimeNanos = ((DefaultRequestLog) log).requestStartTimeNanos;
+                requestStartTimeMillis = log.requestStartTimeMillis();
+            } else {
+                requestStartTimeNanos = System.nanoTime();
+                requestStartTimeMillis = System.currentTimeMillis();
+            }
+            startRequest0(requestStartTimeNanos, requestStartTimeMillis, log.channel(),
+                          log.sessionProtocol(), log.host(), true);
+        }, REQUEST_START);
+        child.addListener(log -> serializationFormat(log.serializationFormat()), SCHEME);
+        child.addListener(log -> requestHeaders(log.requestHeaders()), REQUEST_HEADERS);
+        child.addListener(log -> requestContent(log.requestContent(), log.rawRequestContent()),
+                          REQUEST_CONTENT);
+        child.addListener(log -> {
+            final long requestEndTimeNanos;
+            if (log instanceof DefaultRequestLog) {
+                requestEndTimeNanos = ((DefaultRequestLog) log).requestEndTimeNanos;
+            } else {
+                requestEndTimeNanos = System.nanoTime();
+            }
+
+            if (log.requestCause() != null) {
+                endRequest0(requestEndTimeNanos, log.requestCause());
+            } else {
+                endRequest0(requestEndTimeNanos, null);
+            }
+        }, REQUEST_END);
+    }
+
+    @Override
+    public void endResponseWithLastChild() {
+        checkState(!hasLastChild, "last child is already added");
+        checkState(!children.isEmpty(), "at least one child should be already added");
+        hasLastChild = true;
+        final RequestLog lastChild = children.get(children.size() - 1);
+        propagateResponseSideLog(lastChild);
+    }
+
+    private void propagateResponseSideLog(RequestLog lastChild) {
+        // update the available logs if the lastChild already has them
+        if (lastChild.isAvailable(RequestLogAvailability.RESPONSE_START)) {
```

Comment:
RequestLogAvailability.RESPONSE_START -> RESPONSE_START? Because we have static imports already.

Reply:
Oops~ fixed